### PR TITLE
[Mellanox] Backporting reboot cause to 201811

### PR DESCRIPTION
**Pull request by stephenxs**
**- What I did**
Backporting reboot cause supporting to 201811

**- How I did it**
1. backport the new platform API chassis part to 201811, prune unnecessary code
2. modify makefiles to install sonic_platform on host
3. fix a minor issue in process-reboot-cause

**- How to verify it**
reboot the DUT and then show reboot-cause, check whether the correct reboot cause is displayed.

**- Description for the changelog**
[Mellanox] Backporting reboot cause to 201811

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: files/image_config/process-reboot-cause/process-reboot-cause [hunk 1]
```diff
@@ -70,7 +70,7 @@ def main():
     # if there is no sonic_platform package installed, we only provide
     # software-related reboot causes.
     try:
-        import sonic_platform
+        import sonic_platform.platform
```
Comment (jleveque) [inline files/image_config/process-reboot-cause/process-reboot-cause]: @stephenxs: Why did you need to make this change? This is breaking support for other platforms? I plan to open a PR to revert this line, but I'd like to know your reasoning behind it.

Comment (stephenxs) [reply files/image_config/process-reboot-cause/process-reboot-cause]: I've faced the same situation mentioned in [PR 3156](https://github.com/Azure/sonic-buildimage/pull/3156).
Having read it, I think your suggestion can work.

Comment (jleveque) [reply files/image_config/process-reboot-cause/process-reboot-cause]: Thanks for the reply. My PR to revert this line is here: https://github.com/Azure/sonic-buildimage/pull/3249

Comment (stephenxs) [reply files/image_config/process-reboot-cause/process-reboot-cause]: Thank you. I'll update our code updated today.
